flang/lib/Semantics/mod-file.cpp

```diff
@@ -244,6 +244,14 @@ bool ModFileWriter::PutComponents(const Symbol &typeSymbol) {
   }
 }
 
+// Return the symbol's attributes that should be written
+// into the mod file.
+static Attrs getSymbolAttrsToWrite(const Symbol &symbol) {
+  // Is SAVE attribute is implicit, it should be omitted
+  // to not violate F202x C862 for a common block member.
+  return symbol.attrs() & ~(symbol.implicitAttrs() & Attrs{Attr::SAVE});
```

Comment:
This should apply to all implicit attributes, not just `SAVE`, yes?

Reply:
I am not sure about this.  For example:
```
module equiv
  private
  real :: p
end module
```

`p` gets an implicit `PRIVATE` attribute resulting in this mod file:
```
module equiv
real(4),private::p
end
```

If I filter it out, I will get this:
```
module equiv
real(4)::p
end
```

Reply:
Ok, good point; ship it.